zapisy/apps/grade/poll/utils.py

```diff
@@ -69,34 +70,36 @@ def group_submissions(submissions: List[Submission]) -> dict:
     return grouped_submissions
 
 
-def group(entries: List[Poll], sort=False) -> dict:
-    """Groups a list of polls/submissions into a dictionary.
+def group_polls(entries: List[Poll]) -> Dict[str, Dict[str, List[Poll]]]:
+    """Groups a list of polls into a dictionary of dictionaries.
 
-    The polls and submissions are combined into a dictionary of nested
-    categories and original entries.
+    The polls are grouped by their category into a dictionary of dictionaries,
+    and those nested dictionaries contain grouping by subcategory of the polls
+    within a category.
 
-    This method is structuring data that allows for easy displaying
-    handly tables in views such as the one responsible for summarizing
-    the results of students' submissions.
+    The iteration order of keys of the returned dictionary matters - firstly,
+    the categories of general polls are at the beginning, and secondly, the
+    categories are sorted alphabetically.
     """
-    grouped_entries = defaultdict(list)
-    output = defaultdict(list)
+    def group_polls_by_key(polls: List[Poll], key_extractor: Callable[[Poll], str]) -> Dict[str, List[Poll]]:
+        return {key: list(group) for key, group in groupby(sorted(polls, key=key_extractor),
+                                                           key_extractor)}
+
+    def extract_categories(polls: List[Poll], is_general: bool) -> Set[str]:
+        polls = filter(lambda p: (p.type == PollType.GENERAL) == is_general, polls)
+        return set(map(lambda p: p.category, polls))
```

Comment:
To w końcu struktury składane są fajne czy nie?
```suggestion
        return { p.category for p in polls if p.type == PollType.GENERAL }
```